uktrade/data-hub-leeloo
# Add preview page to import interactions tool

### Description of change

This adds a preview page to the admin site import interactions tool. This displays:

* counts of the loaded rows by contact matching status
* the first 100 rows that will be imported

If no interactions were matched to contacts, a slightly different template is used.

(The actual saving will be added in an upcoming PR.)

[(Previous related PRs.)](https://github.com/uktrade/data-hub-leeloo/pulls?q=is%3Apr+is%3Aclosed+label%3A%22import+interactions%22)

Any suggestions for how to simplify some of the `blocktrans` blocks in the templates?

### To test

1. Create the `admin-interaction-csv-importer` feature flag and set it as active
1. Navigate to the interaction change list page in the admin site
1. Click on the 'Import interaction' link
1. Create a valid or invalid CSV file according to the spec on the page
1. Select your CSV file
1. Submit the form

Example of a valid CSV file (replacing the `team_1`, `adviser_1` and `contact_email` values as necessary):

```
team_1,adviser_1,adviser_2,contact_email,communication_channel,service,kind,date,event_id,notes
Digital Data Hub - Live Service,<an adviser name>,,fooimport@bar.com,Email/Website,Account managemenT,interaction,22/02/2019,,some notes
```

Note that if debug mode is on, it can be slow to load large files. I would suggest turning debug mode off if testing with more than a few dozen rows.

### Screenshots

#### Ready to import interactions

![image](https://user-images.githubusercontent.com/12693549/57461226-88498780-726e-11e9-9c1a-e7e182656836.png)

#### No matched contacts

![image](https://user-images.githubusercontent.com/12693549/57461328-b9c25300-726e-11e9-88be-9b96053f6b0b.png)


### Checklist
